src/arbitrary.rs

```diff
@@ -936,34 +940,58 @@ mod test {
     use std::num::Wrapping;
     use std::path::PathBuf;
     use super::Arbitrary;
+    use super::StdGen;
 
     #[test]
     fn arby_unit() {
         assert_eq!(arby::<()>(), ());
     }
 
+    macro_rules! arby_int {
+        ($($t:ty),+) => {$(
+            let mut gen = super::StdGen::new(rand::thread_rng(), 100);
+            // 1/10 chance of hitting problematic value * 10000 random values
+            // => ~1000 problematic values
+            // => probability of failure ~ (2/3)^(1000) ~ 10e-170
+            let arbys:Vec<$t> = (0..10000)
+                .map(|_|{Arbitrary::arbitrary(& mut gen)})
+                .collect();
+            if !arbys.iter().any(|&x| x==<$t>::max_value()) {
+                panic!("Arbitrary does not generate max value")
+            }
+            if !arbys.iter().any(|&x| x==<$t>::min_value()) {
+                panic!("Arbitrary does not generate min value")
+            }
+            //filter out problematic values
+            let arbys:Vec<$t> = arbys
+                .into_iter()
+                .filter(|x| unsigned_problem_values!($t).iter().any(|y| y==x))
+                .collect();
+            // 9/10 chance of hitting non-problematic value * 10000 random values
+            // => ~9000 values
+            // => probability of failure ~ (19/20)^(9000) ~ 10e-200
+            if  <$t>::max_value()/20 * 19 > *arbys.iter().max().unwrap_or(&0) {
+                panic!("Arbitrary did not generate within 5% of maximum value")
```

Comment:
I rather dislike this method of testing -- simply checking if the highest value is within 5% of the maximum seems fallible.